mlir/include/Driver/Pipelines.h

```diff
@@ -0,0 +1,61 @@
+// Copyright 2024 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "mlir/Pass/Pass.h"
+
+namespace catalyst {
+namespace driver {
+
+void createEnforceRuntimeInvariantsPipeline(mlir::OpPassManager &pm);
+void createHloLoweringPipeline(mlir::OpPassManager &pm);
+void createQuantumCompilationPipeline(mlir::OpPassManager &pm);
+void createBufferizationPipeline(mlir::OpPassManager &pm);
+void createLLVMDialectLoweringPipeline(mlir::OpPassManager &pm);
+void createDefaultCatalystPipeline(mlir::OpPassManager &pm);
+
+void registerEnforceRuntimeInvariantsPipeline();
+void registerHloLoweringPipeline();
+void registerQuantumCompilationPipeline();
+void registerBufferizationPipeline();
+void registerLLVMDialectLoweringPipeline();
+void registerDefaultCatalystPipeline();
+void registerAllCatalystPipelines();
+
+/// Pipeline descriptor
+struct Pipeline {
+    using Name = std::string;
+    using PassList = llvm::SmallVector<std::string>;
+    using PipelineFunc = void (*)(mlir::OpPassManager &);
+    Name name;
```

Comment:
Minor: it's more preferable to have private data fields and structured interfacing methods (so getters, setters, `addPass(...)`), as opposed to have free-floating data fields that can be accidentally accessed.

https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-private

Reply:
I think this struct was first written to just be a small container with limited use. That's why it doesn't have setters and getters. I am not sure if it used enough right now to add those methods but I guess it may get used more later, so I will make it a class.